src/main/java/com/echobox/api/linkedin/types/urn/URN.java

```diff
@@ -20,16 +20,88 @@
 import com.echobox.api.linkedin.jsonmapper.LinkedIn;
 
 import lombok.Getter;
+import lombok.Setter;
 
 /**
- * URN base class - all URNs will contain a URN
- * @author joanna
- *
+ * The type Urn.
+ * @author clementcaylux 
  */
-public abstract class URN {
-  
+public class URN {
+
   @Getter
-  @LinkedIn("$URN")
-  private String urn;
+  @Setter
+  @LinkedIn
+  private String entityType;
+
+  @Getter
+  @Setter
+  @LinkedIn
+  private String id;
+
+  /**
+   * Instantiates a new Urn.
+   *
+   * @param entityType the entity type 
+   * @param id the id
+   */
+  public URN(String entityType, String id) {
+    this.entityType = entityType.toUpperCase();
+    this.id = id;
```

Comment:
Should have null check validation here, as it is not checked in the toString and equals methods